cocotb/handle.py

```diff
@@ -629,7 +657,10 @@ def setimmediatevalue(self, value):
 
     def _getvalue(self):
         binstr = simulator.get_signal_val_binstr(self._handle)
-        result = BinaryValue(binstr, len(binstr))
+        # Skip BinaryValue.assign() as we know we are using a binstr
+        result = BinaryValue(n_bits=len(binstr))
+        # Skip the permitted characters check as we trust the simulator
+        result.set_trusted_binstr(binstr)
```

Comment:
This assumes that we can always trust the string the simulator returns from `get_signal_val_binstr()` only contains permitted characters.
The speed-up was quite significant, so I hope that's a valid assumption.

Reply:
```suggestion
        # Skip the permitted characters check as we trust the simulator
        result._set_trusted_binstr(binstr)
```